Content.Server/Singularity/Components/SinguloFoodComponent.cs

```diff
@@ -9,5 +9,13 @@ public sealed partial class SinguloFoodComponent : Component
         [ViewVariables(VVAccess.ReadWrite)]
         [DataField("energy")]
         public float Energy { get; set; } = 1f;
+
+        /// <summary>
+        /// The percentage of the singularity's current energy that this food will drain.
+        /// Only applies when Energy is negative.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("percentageDrain")]
+        public float PercentageDrain { get; set; }
```

Comment:
```suggestion
        [DataField]
        public float PercentageDrain = 0f;
```
```suggestion
        [DataField]
```
The name is converted automatically from PascalCase to camelCase. Datafield includes VV(RW) by default. Getters and setters are not needed for component classes.
You can do the same for the other datafield above.